### PR TITLE
Feature/recently favorited shelf

### PR DESCRIPTION
**Pull request by vikram-86**
# Why?

We are going to add a shelf on front page 
which contains recently favorited items.. this implements the components according to design
Skjermbilde 2021-11-02 kl  15 11 32

# What?

Adds new cell and a collection view that uses compositional layout

# Version Change

minor

# UI Changes

_If the change concerns UI, a transition or an animation, providing an image, a gif, or a video is encouraged to help us reviewing your PR._

| Before | After |

Simulator Screen Shot - iPhone 12 Pro - 2021-11-02 at 15 12 08

### CODE REVIEW

File: FinniversKit/Sources/Cells/RecentlyFavoritedShelfCell/RecentlyFavoritedShelfCell.swift [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.backgroundColor = .clear
+        label.textAlignment = .center
+        
+        return label
+    }()
+    
+    private lazy var priceBackground: UIView = {
+        let background = UIVisualEffectView(withAutoLayout: true)
+        if #available(iOS 13.0, *) {
```
Comment (osanoj) [inline FinniversKit/Sources/Cells/RecentlyFavoritedShelfCell/RecentlyFavoritedShelfCell.swift:46]: We only support iOS 13 and above now, so this check can be removed 🙂

File: FinniversKit/Sources/Cells/RecentlyFavoritedShelfCell/RecentlyFavoritedShelfCell.swift [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+            background.effect = UIBlurEffect(style: .systemThinMaterialDark)
+        } else {
+            background.effect = nil
+            background.backgroundColor = UIColor.priceLabelBackgroundColor.withAlphaComponent(0.8)
+        }
+        background.alpha = 1.0
+        background.layer.cornerRadius = RecentlyFavoritedShelfCell.priceTagHeight / 2
+        background.clipsToBounds = true
+        
+        background.contentView.addSubview(priceLabel)
```
Comment (osanoj) [inline FinniversKit/Sources/Cells/RecentlyFavoritedShelfCell/RecentlyFavoritedShelfCell.swift:56]: Could you move this and the setup of the constraints into the `setup()` func? To gather all the view configurations there if possible.

File: FinniversKit/Sources/Cells/RecentlyFavoritedShelfCell/RecentlyFavoritedShelfCell.swift (hunk 1 above)
Comment (vikram-86) [reply FinniversKit/Sources/Cells/RecentlyFavoritedShelfCell/RecentlyFavoritedShelfCell.swift:46]: Thats right! totally forgot that part, thanks ! ☺️

File: FinniversKit/Sources/Cells/RecentlyFavoritedShelfCell/RecentlyFavoritedShelfCell.swift (hunk 2 above)
Comment (vikram-86) [reply FinniversKit/Sources/Cells/RecentlyFavoritedShelfCell/RecentlyFavoritedShelfCell.swift:56]: will do! 😁

File: FinniversKit/Sources/Cells/RecentlyFavoritedShelfCell/RecentlyFavoritedShelfCell.swift [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    private var isFavorited: Bool = true {
+        didSet {
+            favoriteButton.isToggled = isFavorited
+        }
+    }
+    
+    private var task: URLSessionDataTask?
+    
+    private lazy var imageView: UIImageView = {
+        let imageView = UIImageView()
```
Comment (osanoj) [inline FinniversKit/Sources/Cells/RecentlyFavoritedShelfCell/RecentlyFavoritedShelfCell.swift:24]: I think using RemoteImageView should be better here, then you do not need any `URLSessionDataTask` in this class and can use the image loading method provided

**Review by osanoj — APPROVED** [review]
LGTM 👍  But I think CircleCI wants a new snapshot image 😄

**Pull request closed without merging**